### PR TITLE
Use CoreDataTestCase for all test cases which need a Core Data stack

### PR DESCRIPTION
**Pull request by crazytonyli**
Follow up PR of https://github.com/wordpress-mobile/WordPress-iOS/pull/18578.

This PR makes changes to lots of files, but they are all similar changes: updating test cases that need a `ContextManagerMock` to inherit from `CoreDataTestCase`, and use `CoreDataTestCase.contextManager` instead of creating their own ones.

## Test Instructions

Make sure all tests pass.

## Regression Notes

N/A. Only unit test target is affected.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/WordPressTest/AccountServiceTests.swift
Comment (mokagio) [inline WordPress/WordPressTest/AccountServiceTests.swift]: Correct me if I'm wrong: Once #18579 lands, we can come back and further DRY this and other tests by using the new `useAsSharedInstance(untilTestFinished:)` method here and removing the `contextManager.tearDown()` call below. Right?

File: WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift
Comment (mokagio) [inline WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift]: Once all your work is done, it might be worth to try removing this `Z`-prefix workaround and see if the issue is still there.

I'm dubious it will help much, because I have a feeling (a true feeling, not an informed suspicion, so I could be very wrong) the reason has more to do with how the production code interacts with Core Data rather than the tests. But, still worth a shot.

File: WordPress/WordPressTest/NotificationSyncMediatorTests.swift
Comment (mokagio) [inline WordPress/WordPressTest/NotificationSyncMediatorTests.swift:84]: I wonder if this commented test could be removed instead...

🤔 Let's just say it's out of scope for this PR and that it's okay to change the code so, if we'll uncomment it, it's more likely it'll work out of the box.

File: WordPress/WordPressTest/ReaderStreamViewControllerTests.swift
Comment (mokagio) [inline WordPress/WordPressTest/ReaderStreamViewControllerTests.swift]: Nitpick. Now that we inherit from `CoreDataTestCase`, this could become:

```suggestion
        let context = mainContext
```

Right?

File: WordPress/WordPressTest/WPAccount+LookupTests.swift
Comment (mokagio) [inline WordPress/WordPressTest/WPAccount+LookupTests.swift]: Might as well remove the `final` while we're at it:

```suggestion
class WPAccountLookupTests: CoreDataTestCase {
```

I don't have the details at hand, but if I recall correctly, there was a time in which marking test cases as `final` helped with compilation speed. Unless I'm mistaken, that's no longer the case, so these can be avoided.

**Review by mokagio — APPROVED** [review]
I left a couple of nitpick suggestions. I was going to write that they don't need to be addressed before merging, but I noticed that there now are merge conflicts against `trunk`, so maybe they can get it if you think they are valid.

Regardless, this is good to go pending merge conflict resolution.

File: WordPress/WordPressTest/AccountServiceTests.swift
Comment (crazytonyli) [reply WordPress/WordPressTest/AccountServiceTests.swift]: Yes, this line [was changed](https://github.com/wordpress-mobile/WordPress-iOS/pull/18579/files#diff-58a0cd66170c21c6183ad4c4347ac79df6238204d98daae8b014fb29d1ecd397) in https://github.com/wordpress-mobile/WordPress-iOS/pull/18579

File: WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift
Comment (crazytonyli) [reply WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift]: I can definitely give it a go 👍

File: WordPress/WordPressTest/AccountServiceTests.swift
Comment (mokagio) [reply WordPress/WordPressTest/AccountServiceTests.swift]: 🎉

File: WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift
Comment (crazytonyli) [reply WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift]: Removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/18660

**Pull request merged**
